relays/client-substrate/src/chain.rs

```diff
@@ -58,7 +58,7 @@ pub trait Chain: ChainBase + Clone {
 	///
 	/// The chain may suport multiple tokens, but this particular type is for token that is used
 	/// to pay for transaction dispatch, to reward different relayers (headers, messages), etc.
-	type Balance: Parameter + Member + DeserializeOwned + Clone + Copy + CheckedSub + PartialOrd + Zero;
+	type Balance: Parameter + Member + DeserializeOwned + Clone + Copy + CheckedSub + PartialOrd + Saturating + Zero;
```

Comment:
According to [the docs](https://docs.rs/num-traits/0.2.14/num_traits/ops/saturating/trait.Saturating.html) the `Saturating` trait is deprecated. We should use the others (`SaturatingAdd`, `SaturatingSub`, etc.) as suggested